### PR TITLE
Use the CompilerFacade form the IntegrationTests

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Use directly the Facade in order to compile phel code into the expected PHP code.

## 🔖 Changes

- Extract to its usage level (on the `Runtime`) the `file_get_content()` from the `FileCompiler::compile()`, so this function will be agnostic from the source itself (former filename) but now rather it receives the "phelCode" to be compiled to PHP.
- Rename `FileCompiler` to `CodeCompiler`.
- Minor cleanup, changing the argument order from `CompilerFacade::lexString()`

## 💡 Follow up

`EvalCompiler` and `CodeCompiler` seem pretty similar. I will try to figure out a way how to combine them if possible in another iteration.

### CODE REVIEW

File: src/php/Runtime/Runtime.php [hunk 1]
```diff
@@ -204,7 +204,8 @@ private function loadCachedFile(string $filename, string $ns): bool
      */
     protected function loadFile(string $filename, string $ns): void
     {
-        $code = $this->compilerFacade->compile($filename);
+        $content = file_get_contents($filename);
+        $code = $this->compilerFacade->compile($content, $filename);
```
Comment (JesusValera) [inline src/php/Runtime/Runtime.php:208]: Makes sense to pass the `filename` in both methods?

Comment (Chemaclass) [reply src/php/Runtime/Runtime.php:208]: Yes. The `compiler()' second argument is the source of the content, it's something required/needed by the lexer in order to identify the source of such, but it's independent of the content itself. This way, the source it's actually an optional argument (being `string` the default value).

File: tests/php/Integration/Compiler/Reader/ReaderTest.php [hunk 2]
```diff
@@ -758,10 +759,10 @@ public function testShortFnRestArgumentMultipleTimes(): void
     /**
      * @return TypeInterface|string|float|int|bool|null
      */
-    private function read(string $string, bool $removeLoc = false)
+    private function read(string $string, bool $withLocation = false)
```
Comment (jenshaase) [inline tests/php/Integration/Compiler/Reader/ReaderTest.php:762]: Shouldn't the the default argument of `$withLocation` also change to `true`

File: tests/php/Integration/Fixtures/Call/global-call.test [hunk 3]
```diff
@@ -12,12 +12,12 @@ $GLOBALS["__phel"]["user"]["x"] = new class() extends \Phel\Lang\AbstractFn {
 };
 $GLOBALS["__phel_meta"]["user"]["x"] = \Phel\Lang\TypeFactory::getInstance()->persistentMapFromKVs(
   new \Phel\Lang\Keyword("start-location"), \Phel\Lang\TypeFactory::getInstance()->persistentMapFromKVs(
-    new \Phel\Lang\Keyword("file"), "Call/global-call.test",
+    new \Phel\Lang\Keyword("file"), "string",
```
Comment (jenshaase) [inline tests/php/Integration/Fixtures/Call/global-call.test:15]: Could we keep the filenames here and in every other test?

**Review by jenshaase — CHANGES_REQUESTED** [review]

File: tests/php/Integration/Compiler/Reader/ReaderTest.php (hunk 2 above)
Comment (Chemaclass) [reply tests/php/Integration/Compiler/Reader/ReaderTest.php:762]: Yes, why not. That thing + changing the line below where this is used. I will do that.
```php
$tokenStream = $this->compilerFacade->lexString($string, Lexer::DEFAULT_SOURCE, $withLocation); # <------
```

File: tests/php/Integration/Fixtures/Call/global-call.test (hunk 3 above)
Comment (Chemaclass) [reply tests/php/Integration/Fixtures/Call/global-call.test:15]: With the first implementation that I did, it wasn't possible, but I just realize that the second option (the one that it's pushed in this PR) is actually possible and super easy, of course! I will change this 💡

**Review by jenshaase — APPROVED** [review]

**Comment by jenshaase** [comment]
Thanks. I merged it.

**Pull request merged**